qiskit/execute_function.py

```diff
@@ -324,8 +324,7 @@ def execute(
 
     if isinstance(backend, BaseBackend):
         # assembling the circuits into a qobj to be run on the backend
-        if shots is None:
```

Comment:
I do not think removing this is a good idea. What one should rather do is that one should add another check of if the backend is DM Simulator, and only in that case should one keep shots None.

Reply:
Yes, I think we already specify the maximum shots limit for `dm_simulator` as 1.